### ISSUE TITLE
Allow user to use a dead key as the prefix key if they want to

### ISSUE THREAD

**Issue by dyavc** [Dyalog/ride#329]
### Describe the issue you are having
Ride 4.0 on macOS, Dyalog 16.0.  User would actually like to use a dead key as the prefix key for typing APL characters.

He describes the dead keys as Orange and cannot be used.

He says:  "In that context, don't you find it a bit ironic that you do not take advantage of "dead keys"? After all they were created to "set a new state that modifies the output of the following keystroke", which is, pretty much, what a definition that would fit the goal of your IDE "prefix key". Is it not?"

**Comment by e9gille** [comment]
I tested doing this last week and it caused headache. Nothing prevents you from defining a dead key as prefix, it does work, but not always the way you'd expect.
For one, the list of suggestions doesn't appear on Windows as it waits for the second keystroke before inserting a character into the editor. On Mac it does, but causes other issues as it replaces the dead key with the new symbol ones you hit the second key.

I might get it to work reliably, but would need to put more of an effort figuring out a good approach.

**Comment by jayfoad** [comment]
My hunch is that this is an unavoidable limitation of the technology we're using. RIDE is based on an embedded Chromium browser, and relies on getting keystroke events from the browser, even for the prefix key. The problem with dead keys is that (I thought) they tend not to generate keystroke events.

But @e9gille you said "it does work" so I guess I'm wrong, and you *do* get keystroke events even for dead keys?

**Comment by e9gille** [comment]
yes, the browser events are triggered for dead keys, so that in itself is not an issue.
